modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -48,5 +54,45 @@ object ConstructionService {
 
       NonEmptyList.fromList(accountIdentifiers)
     }
+
+    def parseSignedTransaction(hex: Hex): EitherT[F, ConstructionError, (Option[NonEmptyList[Operation]], Option[NonEmptyList[Address]], Option[NonEmptyList[AccountIdentifier]])] =
+      KryoSerializer[F]
+        .deserialize[Signed[Transaction]](hex.toBytes)
+        .liftTo[F]
+        .flatMap { signedTransaction =>
+          signedTransaction.proofs.toNonEmptyList
+            .traverse(_.id.hex.toPublicKey.map(_.toAddress))
+            .map(_.foldLeft(List.empty: List[Address]) {
+              case (first, second) =>
+                first.appended(second)
+            })
+            .map(NonEmptyList.fromList(_)).map { maybeAddresses => 
+              val positiveOperation = Operation(
+                OperationIdentifier(OperationIndex(1L)),
+                none,
+                OperationType.Transfer,
+                none,
+                AccountIdentifier(signedTransaction.value.destination, none).some,
+                Amount(AmountValue(signedTransaction.value.amount.value.value), DAGCurrency)
+              )
+
+              val negativeOperation = Operation(
+                OperationIdentifier(OperationIndex(0L)),
+                none,
+                OperationType.Transfer,
+                none,
+                AccountIdentifier(signedTransaction.value.source, none).some,
+                Amount(AmountValue(-1L * signedTransaction.value.amount.value.value), DAGCurrency)
```

Comment:
The only way I know to refine a value at runtime is with `refineV`, but that returns an `Either[...]`. What is the cleanest way to do this?